### ISSUE TITLE
🐛 HTML format result can produce visual difference in Firefox

### ISSUE THREAD

**Issue by tats-u** [biomejs/biome#7304]
### Environment information

```block
Online playground ([2f9d7c9](https://github.com/biomejs/biome/tree/2f9d7c9476f2cfd5eee865ea16d5ba4456ceaa55))
```

### What happened?

https://biomejs.dev/playground/?code=PABkAGkAdgAgAGwAYQBuAGcAPQAiAHoAaAAtAEgAYQBuAHQALQBUAFcAIgA%2BAAoAIAAgABmQC1C1az2EL2ajkLyed5UM%2FwoAKFcATkyI61sNTkyIAjAAZ31ZCgAodQlOTIjrWwIwCgA8AC8AZABpAHYAPgA%3D&language=html

Input:

```html
<div lang="zh-Hant-TW">
  這個段落是那麼長，
在一行寫不行。最好
用三行寫。
</div>
```

Output:

```html
<div lang="zh-Hant-TW">這個段落是那麼長， 在一行寫不行。最好 用三行寫。</div>
```

### Expected result

New lines surrounded only Chinese/Japanese characters should be kept as are as for the time being:

```html
<div lang=zh-Hant-TW>
  這個段落是那麼長，
在一行寫不行。最好
用三行寫。
</div>
```

https://drafts.csswg.org/css-text-4/#line-break-transform

> In languages that have no word separators, such as Chinese, “unbreaking” a line requires joining the two lines with no intervening space.
>
> <div lang=zh-TW>
>
>> 這個段落是那麼長，\
>> 在一行寫不行。最好\
>>用三行寫。
>
>> 這個段落是那麼長，在一行寫不行。最好用三行寫。
> </div>
>
> Figure 7: Eliminating a line break in Chinese requires eliminating any intervening [white space](https://drafts.csswg.org/css-text-4/#white-space).
>
> The segment break transformation rules can use adjacent context to either transform the segment break into a space or eliminate it entirely.

https://codepen.io/tats-u/pen/GgpGpZQ

Only Firefox does not insert spaces there:

<img width="850" height="205" alt="Image" src="https://github.com/user-attachments/assets/347e7a4a-7908-443d-b24c-9560d1ac0e3b" />

On the contrary, Chromium/Safari:

<img width="853" height="89" alt="Image" src="https://github.com/user-attachments/assets/be2116ea-a7f8-4a97-9f55-71dda85a201a" />

Prettier considers this rule to some extent in Markdown, but does not use the same rule as Firefox, so we should not reuse the logic used in Prettier to check the character is Chinese/Japanese.

Prettier (HTML): https://github.com/prettier/prettier/issues/16837
dprint (Markdown): https://github.com/dprint/dprint-plugin-markdown/issues/127

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by tats-u** [comment]
Firefox's Chinese/Japanese character meets all of:

- East Asian Width is F, W, or H
- Script is not Hangul
- Does not have Emoji property
- Not the half-width won sign

A variation selector before the newline is ignored. (Update: [Firefox's implementation is to check the `Default_Ignorable_Code_Point` (`DI`) property of one or more preceding characters](https://github.com/mozilla-firefox/firefox/commit/b1a292990677acae2f2d17ac752d4d22ba978637))

-----

Additional rules in Chinese & Japanse:

If at least one of the sides is CJK punctuations, the space is removed, too. The CJK punctuation meets all:

- East Asian Width is F, W, or H
- Either of:
  - General Category is P and not half-width won sign U+20A9 ₩
  - Is full-width tilde U+FF5E ～
  - (Update) ideographic space U+3000 (since 145)

https://github.com/mozilla-firefox/firefox/blob/66bb0c6d60554576b420b8e7138554c9edcdcc17/intl/unicharutil/util/nsUnicharUtils.cpp#L524 (Caution: the code is MPL!)

https://github.com/w3c/csswg-drafts/issues/5086

**Comment by dyc3** [comment]
Ok, I think I understand the issue. With regards to whitespace sensitivity, we can't add spaces between characters belonging to languages that don't use spaces as word separators. But it _is_ OK to add newlines because the browser accounts for that.

Note this excerpt from the spec:
> Historically, HTML and CSS have unconditionally converted [segment breaks](https://drafts.csswg.org/css-text-4/#segment-break) to spaces, which has prevented content authored in languages such as Chinese from being able to break lines within the source. Thus UA heuristics need to be conservative about where they discard segment breaks even as they strive to improve support for such languages.

I think it makes sense to diverge from prettier for this case.

**Comment by tats-u** [comment]
Web Platform Tests: https://wpt.fyi/results/css/css-text/line-breaking?label=master&label=experimental&aligned&q=segment-break-transformation-

> I think it makes sense to diverge from prettier for this case.

Prettier's behavior should be changed. Biome is going to be just ahead of the curve. We have to report an issue in Prettier's repo but it is difficult to generate an example that is different between the current Prettier and Firefox.

> But it is OK to add newlines because the browser accounts for that.

If "add newlines" means "break the line into two in a Chinese/Japanese line at an arbitrary point", we should avoid it as for now. It causes an extra space there like the above screenshots in unsupported (KHTML/WebKit-derived) browsers. Biome should try to provide the same appearance in all browsers before and after formatting.

---

FYI Prettier's algorithm:

https://github.com/prettier/prettier/blob/main/src/language-markdown/utils.js
https://github.com/prettier/prettier/blob/main/src/language-markdown/constants.evaluate.js
https://github.com/ikatyang-collab/cjk-regex/blob/main/src/index.ts
